### PR TITLE
Test: Added basic unit tests.

### PR DESCRIPTION
**Pull request by chrishogg**
----

**Notes:**

This PR adds basically unit tests to the project. There is also some code refactoring and I've added Serilog.  

[AB#24243](https://dev.azure.com/rezaresystems/15020b92-fee6-4202-956f-9036d1e1da84/_workitems/edit/24243) & 24383

----

## PR Checklist
- [x] Have the changes been tested locally
- [ ] Any instructions necessary have been added
- [ ] The `README.md` file has been update if necessary

### CODE REVIEW

File: src/pfDataSource.Common/CsvBuilder.cs [hunk 1]
```diff
@@ -75,6 +75,11 @@ public Task<bool> WriteAsync(string path)
 				return Task.FromResult(false);
 			}
 		}
+
```
Comment (AndyDe-Aldent) [inline src/pfDataSource.Common/CsvBuilder.cs:78]: Is there a reason we have our own CSV Builder class rather then using a standard CSV package like https://joshclose.github.io/CsvHelper/

File: pfDataSource.Test/DataSourceConfigurationServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+            result.Aws.SecretId.Should().Be("123456789");
+            result.Aws.SecretKey.Should().Be("some-secret-key");
+            result.Configuration.Should().BeNull();
+
+            // Test with configuration object
+            _sourceConfiguration.Configuration = JsonConvert.SerializeObject(new { prop1 = "value1", prop2 = "value2" });
+
+            var result2 = dataSourceConfigurationServiceMock.BuildConfigurationObject(_sourceConfiguration, _sourceConfiguration.AwsSecrectId, _sourceConfiguration.AwsSecretKey, typeof(object));
+
+            result2.Should().NotBeNull();
```
Comment (AndyDe-Aldent) [inline pfDataSource.Test/DataSourceConfigurationServiceTests.cs]: Should these second sets of asserts be seperate test cases?

File: src/pfDataSource.Common/CsvBuilder.cs (hunk 1 above)
Comment (chrishogg) [reply src/pfDataSource.Common/CsvBuilder.cs:78]: ask @daniel-buchanan ;-)

File: pfDataSource.Test/DataSourceConfigurationServiceTests.cs (hunk 2 above)
Comment (chrishogg) [reply pfDataSource.Test/DataSourceConfigurationServiceTests.cs]: Sure, I can pull those out into another case.

File: src/pfDataSource.Common/CsvBuilder.cs (hunk 1 above)
Comment (chrishogg) [reply src/pfDataSource.Common/CsvBuilder.cs:78]: Well update the csv build in another PR

**Review by AndyDe-Aldent — APPROVED** [review]
👍

**Pull request merged**